pkg/cmd/set_replication.go

```diff
@@ -175,19 +187,33 @@ func (o *FinalizeOptions) SetReplication() error {
 		Namespace: o.RepOpts.Source.Namespace,
 		Name:      repSource.Spec.SourcePVC,
 	}
+
 	if err := o.RepOpts.Source.Client.Get(ctx, sourcePVCName, srcPVC); err != nil {
 		return err
 	}
-	repDest := &volsyncv1alpha1.ReplicationDestination{}
-	destNSName := types.NamespacedName{
+
+	var latestImage *corev1.TypedLocalObjectReference
+	repDest = &volsyncv1alpha1.ReplicationDestination{}
+	destNSName = types.NamespacedName{
 		Namespace: o.RepOpts.Dest.Namespace,
 		Name:      o.destName,
 	}
-	if err := o.RepOpts.Dest.Client.Get(ctx, destNSName, repDest); err != nil {
+
+	if err := wait.PollImmediate(5*time.Second, o.timeout, func() (bool, error) {
+		if err := o.RepOpts.Dest.Client.Get(ctx, destNSName, repDest); err != nil {
+			return false, err
+		}
+		latestImage = repDest.Status.LatestImage
+		if oldImage.Name == latestImage.Name {
+			return false, nil
+		}
```

Comment:
- Need to check that `Status` isn't nil
- Need to check that `oldImage` and `latestImage` aren't nil